### PR TITLE
implemented file server

### LINKED ISSUE #1: Create file server
Create a basic server for image storage

### PR DESCRIPTION
**Pull request by FaiyazMujawar**
Fixes #1 

Created a basic file server for saving images.
commit

The data should be sent to `{base_uri}/image` in `multipart/form-data` format. The data should be sent as follows:
```JSON
{
    "userId":"ID of the user",
    "images":"images, if any"
}
```

PS: the order of fields must be same.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const express = require("express");
+
+const app = express();
+
+app.use(express.json());
+
+// Use 'public' folder for serving images
+app.use("/public/images", express.static("public"));
+
+app.use("/", require("./routes/file"));
```
Comment (drigio) [inline index.js]: `/image` route might be better.

File: routes/file.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const router = require("express").Router();
+const saveFile = require("../services/MulterService");
```
Comment (drigio) [inline routes/file.js]: `saveImage` might be a better name. In the future we might have different types of media as well.

File: services/MulterService.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const multer = require("multer");
+const path = require("path");
+
+/** Maximum image count */
+const maxImages = 3;
```
Comment (drigio) [inline services/MulterService.js]: Capitalize constants.

File: services/MulterService.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+const storage = multer.diskStorage({
+  destination: "./public/images",
```
Comment (drigio) [inline services/MulterService.js]: make destination a constant as well.

**Review by drigio — CHANGES_REQUESTED** [review]
Added a few comments. Please have a look.

`multerImageService.js` would be an appropriate name.

File: routes/file.js (hunk 2 above)
Comment (FaiyazMujawar) [reply routes/file.js]: I was thinking of making it generalized. I can change if you say so.

File: index.js (hunk 1 above)
Comment (FaiyazMujawar) [reply index.js]: Okay

**Comment by drigio** [comment]
LGTM! Pulling in changes.

**Review by drigio — APPROVED** [review]

**Pull request merged**